### PR TITLE
add category entity

### PR DESCRIPTION
**Pull request by BohdanTrue**

### CODE REVIEW

**Review by TarasYurkovskyi — APPROVED** [review]
Well done)

File: src/main/java/mate/academy/bookstore/controller/CategoryController.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    @PreAuthorize("hasAnyAuthority('ROLE_USER', 'ROLE_ADMIN')")
+    @GetMapping("/{id}")
+    @Operation(summary = "Get a category by id", description = "Get a category by a certain id")
+    public CategoryResponseDto getCategoryById(@PathVariable Long id) {
+        return categoryService.getById(id);
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/{id}")
```
Comment (d1sam) [inline src/main/java/mate/academy/bookstore/controller/CategoryController.java]:
```suggestion
    @PutMapping("/{id}")
```

File: src/main/java/mate/academy/bookstore/dto/book/BookRequestDto.java [hunk 2]
```diff
@@ -1,25 +1,29 @@
...
-public class CreateBookRequestDto {
+public class BookRequestDto {
     @NotNull(message = "can't be null")
     @Size(min = 3, max = 30)
     private String title;
     @NotNull(message = "can't be null")
     @Size(min = 3, max = 30)
     private String author;
-    @ISBN(message = "Invalid ISBN format")
+    @Pattern(regexp = "^(97(8|9))?\\d{9}(\\d|X)$", message = "Invalid ISBN format")
```
Comment (d1sam) [inline src/main/java/mate/academy/bookstore/dto/book/BookRequestDto.java]: do we need this change in hw related to categories?

File: src/main/java/mate/academy/bookstore/exception/CategoryNotFoundException.java
Comment (d1sam) [inline src/main/java/mate/academy/bookstore/exception/CategoryNotFoundException.java:1]: You can use already existing exception `EntityNotFoundException`.

File: src/main/java/mate/academy/bookstore/model/Category.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+@Where(clause = "is_deleted = false")
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false)
+    private String name;
+
+    private String description;
```
Comment (d1sam) [inline src/main/java/mate/academy/bookstore/model/Category.java:29]:
```suggestion
    @Column
    private String description;
```

File: src/main/java/mate/academy/bookstore/model/Category.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Table(name = "categories")
+@Entity
+@Data
```
Comment (d1sam) [inline src/main/java/mate/academy/bookstore/model/Category.java:16]:
```suggestion
@Data
@NoArgsConstructor
```

File: src/main/resources/db/changelog/changes/12-create-categories-table.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        - createTable:
+            tableName: categories
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/12-create-categories-table.yaml]:
```suggestion
```

File: src/main/resources/db/changelog/changes/13-create-books_categories-table.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      changes:
+        - createTable:
+            tableName: books_categories
+            columns:
+              - column:
+                  name: book_id
+                  type: bigint
+                  constraints:
+                    primaryKey: true
+                    nullable: false
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/13-create-books_categories-table.yaml]:
```suggestion
```

File: src/main/resources/db/changelog/changes/13-create-books_categories-table.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+                    references: books(id)
+                    foreignKeyName: fk_books_categories_book_id
+              - column:
+                  name: category_id
+                  type: bigint
+                  constraints:
+                    primaryKey: true
+                    nullable: false
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/13-create-books_categories-table.yaml]:
```suggestion
```

File: src/main/resources/db/changelog/changes/12-create-categories-table.yaml [hunk 8]
```diff
@@ -0,0 +1,29 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-categories-table
+      author: Bohdan
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/12-create-categories-table.yaml]:
```suggestion
      author: Bohdan Bilko
```

File: src/main/resources/db/changelog/changes/13-create-books_categories-table.yaml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-books_categories-table
+      author: Bohdan
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/13-create-books_categories-table.yaml]:
```suggestion
      author: Bohdan Bilko
```

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but some changes should be done!

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**